src/twisted/protocols/test/test_tls.py

```diff
@@ -1899,108 +1922,111 @@ def dataReceived(self, data):
 
         clientDataReceived = Deferred()
         clientFactory = ClientNegotiationFactory(clientProtocols)
-        clientFactory.protocol = lambda: NotifyingSender(clientDataReceived)
+        clientFactory.protocol = lambda: NotifyingSender(
+            clientDataReceived, b"data from client"
+        )
 
-        clientContextFactory, _ = HandshakeCallbackContextFactory.factoryAndDeferred()
+        clientContextFactory = ClientTLSContext()
+        # See why ClientNegotiationFactory is not working.
+        _setAcceptableProtocols(clientContextFactory.getContext(), clientProtocols)
```

Comment:
this call should not be made here... as ClientTLSContext trigger the automatic call.